core-bundle/tests/Image/Studio/FigureBuilderTest.php

```diff
@@ -324,6 +357,99 @@ public function provideMixedIdentifiers(): \Generator
         yield 'absolute path' => [$absoluteFilePath];
     }
 
+    public function testLastExceptionIsResetWhenCallingFrom(): void
+    {
+        [$absoluteFilePath, $relativeFilePath] = $this->getTestFilePaths();
+
+        /** @var FilesModel&MockObject $model */
+        $model = $this->mockClassWithProperties(FilesModel::class);
+        $model->type = 'file';
+        $model->path = $relativeFilePath;
+
+        /** @var FilesModel&MockObject $invalidModel */
+        $invalidModel = $this->mockClassWithProperties(FilesModel::class);
+        $invalidModel->type = 'folder';
+
+        $filesModelAdapter = $this->mockAdapter(['findByUuid', 'findByPk', 'findByPath']);
+        $filesModelAdapter
+            ->method('findByUuid')
+            ->willReturnMap([
+                ['foo-uuid', $model],
+                ['invalid-uuid', null],
+            ])
+        ;
+
+        $filesModelAdapter
+            ->method('findByPk')
+            ->willReturnMap([
+                [5, $model],
+                [123, null],
+            ])
+        ;
+
+        $filesModelAdapter
+            ->method('findByPath')
+            ->willReturnMap([
+                [$absoluteFilePath, $model],
+                ['invalid/path', null],
+            ])
+        ;
+
+        $framework = $this->mockContaoFramework([FilesModel::class => $filesModelAdapter]);
+        $studio = $this->getStudioMockForImage($absoluteFilePath);
+        $figureBuilder = $this->getFigureBuilder($studio, $framework);
+
+        $setValidResourceOperations = [
+            ['fromFilesModel', $model],
+            ['fromUuid', 'foo-uuid'],
+            ['fromID', 5],
+            ['fromPath', $absoluteFilePath],
+        ];
+
+        $setInvalidResourceOperations = [
+            ['fromFilesModel', $invalidModel],
+            ['fromUuid', 'invalid-uuid'],
+            ['fromID', 123],
+            ['fromPath', 'invalid/path'],
+        ];
+
+        $exception = null;
+
+        foreach ($setInvalidResourceOperations as $operation) {
+            [$method, $argument] = $operation;
+
+            $figureBuilder->$method($argument);
+
+            $this->assertNotSame(
+                $exception,
+                $figureBuilder->getLastException(),
+                'new exception replaces old one'
+            );
+
+            $exception = $figureBuilder->getLastException();
+
+            $this->assertInstanceOf(InvalidResourceException::class, $exception);
+        }
+
+        foreach ($setValidResourceOperations as $operation) {
+            [$method, $argument] = $operation;
```

Comment:
```suggestion
        foreach ($setValidResourceOperations as [$method, $argument]) {
```

Reply:
short and expressive - I like it :slightly_smiling_face:

Reply:
I didn't even know this was possible. 🦊